components/modal/modal-content.directive.ts

```diff
@@ -3,12 +3,12 @@
  * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
  */
 
-import { Directive, TemplateRef } from '@angular/core';
+import { Directive, inject, TemplateRef } from '@angular/core';
 
 @Directive({
   selector: '[nzModalContent]',
   exportAs: 'nzModalContent'
 })
 export class NzModalContentDirective {
-  constructor(public templateRef: TemplateRef<{}>) {}
+  public readonly templateRef = inject(TemplateRef<{}>);
```

Comment:
It seems that inject would lose generic type `<{}>`, the inferred of templateRef is `TemplateRef<any>`, so I think we should declare it's type explicitly: 

```suggestion
  public readonly templateRef: TemplateRef<{}> = inject(TemplateRef);
```

BTW ElementRef has same issue :)